torch/csrc/autograd/custom_function.cpp

```diff
@@ -113,9 +119,9 @@ variable_list _wrap_outputs(const variable_list &input_vars,
 
     if (cdata) {
       auto output_nr = cdata->add_input_metadata(var);
-      AT_ASSERT(i == (int)output_nr);
+      AT_ASSERT(num_tensors == (int)output_nr);
```

Comment:
Both num_tensors and output_nr are `unit32_t`, please remove the cast
```suggestion
      AT_ASSERT(num_tensors == output_nr);
```